rosidl_generator_c/package.xml

```diff
@@ -26,6 +26,7 @@
   <exec_depend>rosidl_parser</exec_depend>
   <exec_depend>rcutils</exec_depend>
 
+  <test_depend>ament_cmake_cppcheck</test_depend>
```

Comment:
Theoretically, this shouldn't be necessary; we already have a dependency on ament_lint_common, and that already contains `ament_cmake_cppcheck`: https://github.com/ament/ament_lint/blob/master/ament_lint_common/package.xml .

But maybe I'm missing something, so can you explain why you think this is necessary?

Reply:
I saw the error in the install job and saw that the line failing is this one:
https://github.com/ros2/rosidl/blob/de0556a13be36ca62d6a6573da369e3fe2eb204c/rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake#L177 and opened the PR to test if it solved it. However, I think the install jobs use the released version, so I don't know if there's an easy way to test this PR.

Yeah, I saw `ament_lint_common` but I'm not sure if it works internally in a different manner for the install jobs. 

I think the PR introducing this regression is https://github.com/ros2/rosidl/pull/662. Tagging @sloretz to see if he has an idea about why this could be happening.

Reply:
From the CI log:

```
+ rosdep install --from-paths . -r --ignore-src --rosdistro=rolling --default-yes
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
gazebo_ros: No definition of [tinyxml_vendor] for OS version [focal]
gazebo_plugins: No definition of [tf2_ros] for OS version [focal]
gazebo_msgs: No definition of [ament_lint_common] for OS version [focal]
```

It looks like `ament_lint_common` didn't get installed.

Reply:
> + rosdep install --from-paths . -r --ignore-src --rosdistro=rolling --default-yes
> ERROR: the following packages/stacks could not have their rosdep keys resolved
> to system dependencies:
> gazebo_ros: No definition of [tinyxml_vendor] for OS version [focal]
> gazebo_plugins: No definition of [tf2_ros] for OS version [focal]
> gazebo_msgs: No definition of [ament_lint_common] for OS version [focal]

Oh, thanks for pointing that out.  Note that this is *never* going to work; while the packages still exist for Rolling-on-Focal, the rosdep database now only serves Rolling-on-Jammy.  So I think that is the main issue.

Reply:
Thanks for the feedback, I'll talk with Jose about how to move forward with this. I imagine we should be moving this job to jammy instead.